src/lib/consensus/mechanism.ml

```diff
@@ -1,154 +1,106 @@
-   * transition cannot be generated.
-   *)
-
-  val is_transition_valid_checked :
-       Snark_transition.var
-    -> (Snark_params.Tick.Boolean.var, _) Snark_params.Tick.Checked.t
-  (**
-   * Create a checked boolean constraint for the validity of a transition.
-   *)
-
-  val next_state_checked :
-       Consensus_state.var
-    -> Coda_base.State_hash.var
-    -> Snark_transition.var
-    -> Currency.Amount.var
-    -> (Consensus_state.var, _) Snark_params.Tick.Checked.t
-  (**
-   * Create a constrained, checked var for the next consensus state of
-   * a given consensus state and snark transition.
-   *)
-
-  val select :
-       existing:Consensus_state.value
-    -> candidate:Consensus_state.value
-    -> logger:Logger.t
-    -> time_received:Unix_timestamp.t
-    -> [`Keep | `Take]
-  (**
-   * Select between two ledger builder controller tips given the consensus
-   * states for the two tips. Returns `\`Keep` if the first tip should be
-   * kept, or `\`Take` if the second tip should be taken instead.
-   *)
-
-  val next_proposal :
-       Unix_timestamp.t
-    -> Consensus_state.value
-    -> local_state:Local_state.t
-    -> keypair:Signature_lib.Keypair.t
-    -> logger:Logger.t
-    -> [ `Check_again of Unix_timestamp.t
-       | `Propose of Unix_timestamp.t * Proposal_data.t ]
-  (**
-   * Determine if and when to perform the next transition proposal. Either
-   * informs the callee to check again at some time in the future, or to
-   * schedule a proposal at some time in the future.
-   *)
-
-  val lock_transition :
-       ?proposer_public_key:Signature_lib.Public_key.Compressed.t
-    -> Consensus_state.value
-    -> Consensus_state.value
-    -> snarked_ledger:(unit -> Coda_base.Ledger.t Or_error.t)
-    -> local_state:Local_state.t
-    -> unit
-  (**
-   * A hook for managing local state when the locked tip is updated.
-   *)
-end
+[%%import
+"../../config.mlh"]
+
+open Core
+open Async
+include Intf
+
+let exit1 ?msg =
+  match msg with
+  | None -> Core.exit 1
+  | Some msg ->
+      Core.Printf.eprintf "%s\n" msg ;
+      Core.exit 1
+
+let env name ~f ~default =
+  let name = Printf.sprintf "CODA_%s" name in
+  Unix.getenv name
+  |> Option.map ~f:(fun x ->
+         match f @@ String.uppercase x with
+         | Some v -> v
+         | None ->
+             exit1
+               ~msg:
+                 (Printf.sprintf
+                    "Inside env var %s, there was a value I don't understand \
+                     \"%s\""
+                    name x) )
+  |> Option.value ~default
+
+[%%if
+defined consensus_mechanism]
+
+[%%if
+consensus_mechanism = "sigs"]
+
+module Make (Ledger_builder_diff : sig
+  type t [@@deriving sexp, bin_io]
+end) =
+Proof_of_signature.Make (struct
+  module Genesis_ledger = Genesis_ledger
+  module Proof = Coda_base.Proof
+  module Ledger_builder_diff = Ledger_builder_diff
+  module Time = Coda_base.Block_time
+
+  let proposal_interval =
+    env "PROPOSAL_INTERVAL"
```

Comment:
Let's log an issue to turn these into configuration variables read from the config file.

Reply:
Made the issue https://github.com/CodaProtocol/coda/issues/1182